packages/caliper-fabric/lib/fabric.js

```diff
@@ -14,2181 +14,57 @@
-                }
-
-                CaliperUtils.assertDefined(queryResponse);
-                CaliperUtils.assertProperty(queryResponse, 'queryResponse', 'chaincodes');
-
-                if (queryResponse.chaincodes.some(
-                    cc => cc.name === chaincodeInfo.id && cc.version === chaincodeInfo.version)) {
-                    logger.info(`${chaincodeInfo.id}@${chaincodeInfo.version} is already instantiated in ${channel}`);
-                    continue;
-                }
-
-                chaincodeInstantiated = true;
-
-                let txId = admin.newTransactionID(true);
-                /** @link{ChaincodeInstantiateUpgradeRequest} */
-                let request = {
-                    targets: targetPeers,
-                    chaincodeId: ccObject.id,
-                    chaincodeVersion: ccObject.version,
-                    chaincodeType: ccObject.language,
-                    args: ccObject.init || [],
-                    fcn: ccObject.function || 'init',
-                    'endorsement-policy': ccObject['endorsement-policy'] ||
-                        this.networkUtil.getDefaultEndorsementPolicy(channel, { id: ccObject.id, version: ccObject.version }),
-                    transientMap: this.networkUtil.getTransientMapOfChaincodeOfChannel(chaincodeInfo, channel),
-                    txId: txId
-                };
-
-                // check chaincode language
-                // other chaincodes types are not supported in every version
-                if (ccObject.language !== 'golang') {
-                    if (ccObject.language === 'node' && this.networkUtil.isInCompatibilityMode()) {
-                        throw new Error(`${chaincodeInfo.id}@${chaincodeInfo.version} in ${channel}: Node.js chaincodes are supported starting from Fabric v1.1`);
-                    }
-
-                    if (ccObject.language === 'java' && this.version.lessThan('1.3.0')) {
-                        throw new Error(`${chaincodeInfo.id}@${chaincodeInfo.version} in ${channel}: Java chaincodes are supported starting from Fabric v1.3`);
-                    }
-
-                    if (!['golang', 'node', 'java'].includes(ccObject.language)) {
-                        throw new Error(`${chaincodeInfo.id}@${chaincodeInfo.version} in ${channel}: unknown chaincode type ${ccObject.language}`);
-                    }
-                }
-
-                // check private collection configuration
-                if (CaliperUtils.checkProperty(ccObject, 'collections-config')) {
-                    if (this.version.lessThan('1.2.0')) {
-                        throw new Error(`${chaincodeInfo.id}@${chaincodeInfo.version} in ${channel}: private collections are supported from Fabric v1.2`);
-                    }
-
-                    request['collections-config'] = ccObject['collections-con
```

Comment:
'fabric-client' being removed, so will not work in future. Use 'fabric-ca-client' instead.

Reply:
good spot 👍

Reply:
latest push bases the conditioning on fabric-ca-client 👍